docs/testplan/BGP-Scale-Test.md

```diff
@@ -0,0 +1,124 @@
+- [Setup Configuration](#Setup-Configuration)
+- [Test Methodology](#Test-Methodology)
+- [Test Cases](#Test-Cases)
+  - [BGP Sessions Flapping Test](#BGP-Sessions-Flapping-Test)
+  - [Unisolation Test](#Unisolation-Test)
+  - [Nexthop Group Member Scale Test](#Nexthop-Group-Member-Scale-Test)
+
+
+# Overview
+
+This test plan is to test if control/data plane can handle the initialization/flapping of numerous BGP session holding a lot routes, and estimate the impact on it.
+
+
+## Scope
+
+This test plan runs on any device running SONIC system with fully functioning configuration with numerouse BGP peers with count 256/512.
+
+This test plan is dedicated to IPv6.
+
+This test plan shows if there is any service crush, if hardware resource run out, if device has acceptable performance and data/control plane availability.
+
+
+## Testbed
+
+This test run on testbeds with topologies:
+- t0: topo_t0-isolated-d2u254s1, topo_t0-isolated-d2u254s2, topo_t0-isolated-d2u510
+- t1: topo_t1-isolated-d254u2s1, topo_t1-isolated-d254u2s2 and topo_t1-isolated-d510u2.
+
+<img src="Img/t0-isolated-d2u254s.png" width="360" height="240">
+
+*Fig.1 topo_t0-isolated-d2u254s1/s2*
+
+<img src="Img/t0-isolated-d2u510.png" width="160" height="240">
+
+*Fig.2 topo_t0-isolated-d2u510*
+
+<img src="Img/t1-isolated-d254u2s.png" width="360" height="240">
+
+*Fig.3 topo_t1-isolated-d254u2s1/s2*
+
+<img src="Img/t1-isolated-d510u2.png" width="160" height="240">
+
+*Fig.4 topo_t1-isolated-d510u2*
+
+
+## Scale Description
+In this scale test, the maximum bgp peers number reach crazy count 510!
+
+We haven't even tried 200 before, so this is a giant leap for us.
+
+With great number of bgp peers, come with the great number of ceos containers that are used to establish bgp sessions with full NOS functionality for all possible testing.
+
+One ceos will comsume around 1.3GB memory, 510 means we need a server with 663GB. To deploy this topology with 510 even more bgp peers, we initial a design to achieve [deploy testbed with multiple servers](https://github.com/sonic-net/sonic-mgmt/pull/15395), with this design, we can leverage multiple servers' CPU, memory and network capacity, to break the single server resource limitation.
+
+# Setup Configuration
+The count of routes from BGP peers is vital, we will leverage exabpg to advertise routes to all BGP peers, and those routes be be advertised to device under test finally.
+
+When DUT is T0, via exabgp, firstly, we will advertise 511 routes with prefix length 120 to all peer T1 devices for simulating downstream routes (VLAN IPv6 addresses of T0s), secondly, we will dvertise 15 routes with prefix length 64 to all peer T1 devices for simulating upstream routes (Aggregated IPv6 addresses of T0s' VLAN on T2s), finally, the DUT T0 will receive those routes from BGP peers.
+
+When DUT is T1, we won't mock any routes.
+
+
+# Test Methodology
+For simulating the initialization of system, we shutdown all ports before test.
```

Comment:
it will be better to list the tests in sections:

```markdown
## Initial state

shutdown all the ports and etc goes here.

## Test cases

### BGP Sessions Flapping Test

shutdown a single port goes here. objective and details steps.

### Unisolation tests and more tests

follow the similar idea as above to explain here
```

Reply:
Updated